src/Splunk.OpenTelemetry.AutoInstrumentation.Plugin/Traces.cs

```diff
@@ -14,22 +14,71 @@
 // limitations under the License.
 // </copyright>
 
+using System;
+using OpenTelemetry.Exporter;
 using OpenTelemetry.Trace;
 
-namespace Splunk.OpenTelemetry.AutoInstrumentation.Plugin;
-
-/// <summary>
-/// Tracer plugin
-/// </summary>
-public class Traces
+namespace Splunk.OpenTelemetry.AutoInstrumentation.Plugin
 {
     /// <summary>
-    /// Configures Traces
+    /// Splunk OTel traces plugin
     /// </summary>
-    /// <param name="builder"><see cref="TracerProviderBuilder"/>  to configure</param>
-    /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
-    public TracerProviderBuilder ConfigureTracerProvider(TracerProviderBuilder builder)
+    public class Traces
     {
-        return builder;
+        private readonly PluginSettings _settings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Traces"/> class.
+        /// </summary>
+        public Traces()
+            : this(PluginSettings.FromDefaultSources())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Traces"/> class.
+        /// This is constructor is for testing purposes.
+        /// </summary>
+        /// <param name="settings">PluginSettins instance</param>
+        internal Traces(PluginSettings settings)
+        {
+            _settings = settings;
+        }
+
+        /// <summary>
+        /// Configures Traces
+        /// </summary>
+        /// <param name="builder"><see cref="TracerProviderBuilder"/>  to configure</param>
+        /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
+        public TracerProviderBuilder ConfigureTracerProvider(TracerProviderBuilder builder)
+        {
+            return builder;
+        }
+
+        /// <summary>
+        /// Configure Otlp exporter options
+        /// </summary>
+        /// <param name="options">Otlp options</param>
+        public void ConfigureOptions(OtlpExporterOptions options)
```

Comment:
We would need to document that direct ingest is supported only when OTLP exporter is used.
Alternatively, we can add support for other exporters, but I think it is YAGNI.